src/test/scala/edu/berkeley/ce/rockslicing/EndToEndSpec.scala

```diff
@@ -17,26 +17,57 @@ class EndToEndSpec extends FunSuite {
     // Create an initial block
     val blocks = Seq(Block(globalOrigin, rockVolume))
 
+    // Generate seed joint
+    val numSeedJoints = 5
+    val seedJoints = LoadBalancer.generateSeedJoints(blocks.head, numSeedJoints)
+    val joints = seedJoints ++ jointList
+
     // Iterate through joints, cutting blocks where appropriate
     var cutBlocks = blocks
-    for (joint <- jointList) {
+    for (joint <- joints) {
       cutBlocks = cutBlocks.flatMap(_.cut(joint))
     }
 
     // Remove geometrically redundant joints
-    val nonRedundantBlocks = cutBlocks.map { case block@Block(center, _) =>
+    val nonRedundantBlocks = cutBlocks.map { case block @ Block(center, _) =>
       Block(center, block.nonRedundantFaces)
     }
 
-    // Calculate the centroid of each block
-    val centroidBlocks = nonRedundantBlocks.map { block =>
+    // Find all blocks that contain processor joints
+    val processorBlocks = nonRedundantBlocks.filter { block => 
```

Comment:
Maybe. I think it would be okay to use `partition`, but if we stick with the two calls to `filter` let's add a comment explaining that we do this to match the Spark code.